adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -153,8 +153,13 @@ func (c *Deserializer) ReadNode(r io.Reader, res *DeserializationResult) error {
 	}
 
 	if res.Nodes[id] == nil {
-		res.Nodes[id] = &vertex{level: int(level), id: id, connections: make(map[int][]uint64)}
+		res.Nodes[id] = &vertex{level: int(level), id: id, connections: make([][]uint64, level+1)}
 	} else {
+		if int(level) >= len(res.Nodes[id].connections) {
+			newConns := make([][]uint64, level+1)
```

Comment:
Why is this level + 1 and not `len(res.Nodes[id].connections)`? You copy `res.Nodes[id].connections` into the buffer, so those need to have the same size?

Reply:
The `level` variable contains the max level, so for example, if the `level` is set to `1`, we have two dimensions that each have connections:  `level 0` and `level 1`, thus we need `len=2`.

The `copy` is just so that the state after the grow operation is the same as before. Since the level was previously lower, we didn't have any connection entries yet. But this could be followed by a `ReadLink` instruction ,for example, which would then try to write at `connections[1] = ...` which requires `len(connections)==2`